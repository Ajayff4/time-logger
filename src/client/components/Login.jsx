import React, { Component } from 'react';
import axios from 'axios';
import sha256 from 'sha256';
import { withRouter } from 'react-router-dom';

class Login extends Component {

    state = {
        username: "",
        password: ""
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value });
    }

    onSubmit = (e) => {
        e.preventDefault();
        const { username, password } = this.state;
        axios.post('/login', {
            username: username,
            password: sha256(password)
        }).then(res => {
            console.log("response: ", res.data);
            this.props.history.push('/logs');
        }).catch(err => {
            console.error("post: ", err);
            //this.props.handleErrors(err);
        });
    }

    render() {
        return (
            <center>
                <div>
                    <fieldset>
                        <legend>Login</legend>
                        <form method="POST" onSubmit={this.onSubmit}>
                            <input
                                type="text"
                                name="username"
                                id="username"
                                placeholder="username"
                                value={this.state.username}
                                onChange={this.onChange}
                                required
                            /><br />
                            <input
                                type="password"
                                name="password"
                                id="password"
                                placeholder="password"
                                value={this.state.password}
                                onChange={this.onChange}
                                required
                            /><br />
                            <button type="submit">Login</button>
                        </form>
                    </fieldset>
                </div>
            </center>
        )
    }
}

export default withRouter(Login);