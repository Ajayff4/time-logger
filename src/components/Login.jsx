import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import * as actions from '../actions'
import PropTypes from 'prop-types'
import axios from 'axios'

class Login extends Component {
    static propTypes = {
        setUserData: PropTypes.func.isRequired,
        failedToFetchData: PropTypes.func.isRequired
    }

    constructor(props) {
        super(props)
        this.state = {
            username: "",
            password: ""
        }
    }

    onChange = (e) => {
        this.setState({ [e.target.name]: e.target.value })
    }

    onSubmit = (e) => {
        e.preventDefault()
        this.props.authStart();
        const {username, password} = this.state;
        axios.post("http://localhost:5000/rest/users/login", {
            username: username,
            password: password
        }).then(res => {
            this.props.setUserData(res.data.username, res.data.fullname, res.data.email)
            this.props.setCookie();
            this.props.history.push('/logs')
            console.log(username, document.cookie.username, username===document.cookie.username)
        }).catch(err => {
            console.log("login", err)
            this.props.failedToFetchData(`${err}`);
        })
    }

    render() {
        return (
            <center>
                <div>
                    <fieldset>
                        <legend>Login</legend>
                        <form method="POST" onSubmit={this.onSubmit} autoComplete="off">
                            <input
                                type="text"
                                name="username"
                                id="username"
                                placeholder="username"
                                value={this.state.username}
                                onChange={this.onChange}
                                required
                            /><br />
                            <input
                                type="password"
                                name="password"
                                id="password"
                                placeholder="password"
                                value={this.state.password}
                                onChange={this.onChange}
                                required
                            /><br />
                            <button type="submit">Login</button>
                        </form>
                    </fieldset>
                </div>
            </center>
        )
    }
}

const mapStateToProps = state => {
    return {
        isLoggedIn: state.user.loggedIn
    }
}

const mapDispatchToProps = dispatch => {
    return {
        authStart: () => dispatch(actions.authStart()),
        setCookie: () => dispatch(actions.setCookie()),
        failedToFetchData: (error) => dispatch(actions.failedToFetchData(error)),
        setUserData: (username, fullname, email) => dispatch(actions.setUserData(username, fullname, email))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(Login))