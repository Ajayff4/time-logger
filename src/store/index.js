import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store