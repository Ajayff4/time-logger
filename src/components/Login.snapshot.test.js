import React from 'react'
import renderer from 'react-test-renderer'
import Login from './Login'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Login Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><Login /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});