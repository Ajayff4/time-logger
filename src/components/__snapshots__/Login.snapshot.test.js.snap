// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshot Testing... renders component correctly 1`] = `
<center>
  <div>
    <fieldset>
      <legend>
        Login
      </legend>
      <form
        autoComplete="off"
        method="POST"
        onSubmit={[Function]}
      >
        <input
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="username"
          required={true}
          type="text"
          value=""
        />
        <br />
        <input
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="password"
          required={true}
          type="password"
          value=""
        />
        <br />
        <button
          type="submit"
        >
          Login
        </button>
      </form>
    </fieldset>
  </div>
</center>
`;
